sig-security-assessments/cluster-api/self-assessment.md

```diff
@@ -0,0 +1,1099 @@
+  performed using cloud credentials granted to the management and workload
+  cluster’s control plane. Additionally, implement centralized audit collection
+  and alerting for any suspicious activity.
+  * Status: [End user guidance to be written](#cluster-api-end-user-guide)
+
+### Information Disclosure
+
+#### STRIDE-INFODISCLOSE-1 Privilege escalation from workload cluster co-located with management cluster
+
+Cluster API Provider AWS by default uses the same set of permissions for
+workload and management control planes, and if a workload cluster is colocated
+in the same AWS account as a management cluster, a user can use the credentials
+in the workload cluster to snapshot the disk of the management cluster, create a
+new EC2 volume, mount it to a node in the workload cluster, and exfiltrate the
+etcd datastore.
+
+#### STRIDE-INFODISCLOSE-1 Recommended mitigations
+
+* CAPA should by default use separate IAM roles for workload clusters to
+  management clusters to reduce the amount of privileges workload clusters have.
+  * Status: Not yet considered
+* Cluster API should consider supporting running CPI, CSI and other privileged
+  control plane workloads on the management cluster to further minimise the
+  privileges required by workload clusters.
+  *
+  Status: [Issue filed](https://github.com/kubernetes-sigs/cluster-api/issues/5491)
+* Strongly advise users to seperate workload clusters into different AWS
+  accounts or Azure subscriptions.
+  * Status: [End user guidance to be written](#cluster-api-end-user-guide)
+
+#### STRIDE-INFODISCLOSE-2 Exposure of kubeadm token credentials
+
+Exposure of kubeadm token located on the node or in Cloud resources like Secrets
+Manager can lead to full or partial control over one or more clusters, and can
+be reused to register as the control plane node itself, after which attacker can
+get access to control plane secrets.
+
+#### STRIDE-INFODISCLOSE-2 Recommended Mitigations
+
+* kubeadm token is short lived or deleted
+  * Status: implemented / planned / to be implemented / End User Guidance
+* Disable or restrict SSH access to nodes in a cluster
+  * Status: implemented / planned / to be implemented / End User Guidance
+* Audit KubeConfig files access, edits that are located on the node
+  * Status: implemented / planned / to be implemented / End User Guidance
+* Implement cluster level Pod Security to prevent host mount access to pod
+  * Status: implemented / planned / to be implemented / End User Guidance
+
+#### STRIDE-INFODISCLOSE-3 - Exposure of Management and Workload Cluster CA private keys and certs
+
+Exposure of Management Cluster CA private keys and certs on the node or in the
+cloud resources like Secrets Manager can lead to full or partial control over
+one or more clusters.
+
+#### STRIDE-INFODISCLOSE-3 Recommended Mitigations
+
+* Keep CA key/cert pairs short lived or deleted after use from Secrets Manager
+* Shorten CA cert lifetime
```

Comment:
just to mention it this requires a solution for CA rotation. Otherwise we have a big problem.

Reply:
Yes